nap/utils.py

```diff
@@ -1,9 +1,36 @@
 from __future__ import unicode_literals
+from collections import namedtuple
 import itertools
+from urlparse import urlparse
+
 import six
 from six.moves import urllib
 
 
+def normalize_url(url_string):
+    QueryParam = namedtuple("QueryParam", "key value")
+    url = urlparse(url_string)
```

Comment:
The docs suggest `urlsplit` over `urlparse` in typical uses. Apparently `urlsplit` supports a newer variation of URL.

https://docs.python.org/2/library/urlparse.html#urlparse.urlsplit

> "This should generally be used instead of urlparse() if the more recent URL syntax ..."